ArduCopter/flight_mode.cpp

```diff
@@ -262,7 +267,7 @@ void Copter::exit_mode(control_mode_t old_control_mode, control_mode_t new_contr
 #endif
 
     // stop mission when we leave auto mode
-    if (old_control_mode == AUTO) {
+    if (old_control_mode == AUTO && control_mode != AUTO) {
```

Comment:
It's needed. Changing from AUTO to AUTO is the common case here. You are in an AUTO mission and trigger an AFS failsafe.